### ISSUE TITLE
Would you support lug.ustc.edu.cn ?

### ISSUE THREAD

**Issue by renyouguo** [Synzvato/decentraleyes#103]
fonts.googleapis.com-------------->fonts.lug.ustc.edu.cn
ajax.googleapis.com--------------->ajax.lug.ustc.edu.cn
themes.googleusercontent.com------>google-themes.lug.ustc.edu.cn
fonts.gstatic.com----------------->fonts-gstatic.lug.ustc.edu.cn

lug.ustc.edu.cn is a very popular mirror site for google in China.
They are equal.

**Comment by Synzvato** [comment]
@renyouguo That's a great catch, thanks for sharing! We should definitely start supporting domain aliases if these sorts of mirror networks exist and have significant amounts of users.

**Comment by CHEF-KOCH** [comment]
I don#t see any improvements, you simply trust other font providers instead of the well known. makes less sense, especially the mentioned are known for years and never compromised, why should i now _trust_ others? 

Don't get it.

**Comment by renyouguo** [comment]
@CHEF-KOCH Most chinese cannot visit google, so there are mirrors in china.
You can see the extension for chrome/firefox: gooreplacer.

**Comment by CHEF-KOCH** [comment]
[Bullshit](https://en.wikipedia.org/wiki/Google_China), it's number 3 of the internet machines of china. One of Googles headwaters stands in china. Font's and 'geoblocking' have nothing much to do with each other. Sorry, check the facts before you post.

**Comment by Synzvato** [comment]
@CHEF-KOCH

I think @renyouguo's assessment is spot on. Google's market share in China is small compared to its main competitors _(1.7%)_. As can be seen on the Wikipedia page you linked to:

> Google has had a rocky relationship with the Chinese authorities since January 2010 [...] In 2014 [...] China made all Google services almost unusable [...] In 2009, one-third of all searches in China were on Google. Now the US company has only 1.7% market share.

In addition, most of the domains @renyouguo suggested, including ```fonts.googleapis.com``` _(Google Fonts)_, are being actively blocked in mainland China. See the link below for more details:

https://en.wikipedia.org/wiki/Websites_blocked_in_mainland_China

**Comment by CHEF-KOCH** [comment]
It's still wrong. Marketshare is because pre-installed engines (similar like IE on Windows or if you want to m compare on Browser level Bing/Yahoo/..).

The articles says at the very end, it's because the government firewall (great canon). It's not even related to google. Set a proxy as china and you will see Google works.Even if fonts are blocked you still can use it. [It's mentioned here](https://blog.torproject.org/blog/closer-look-great-firewall-china) what was _blocked_.  It's not Googles firewall it's government and since no one here seems to like google (for whatever reasons) it doesn#t even matter since you simply can use DuckDuck Go (uses Yahoo + some kind of Google Index) or alternatives, they working even in China.

**Comment by Synzvato** [comment]
> It's still wrong. Marketshare is because pre-installed engines (similar like IE on Windows or if you want to compare on Browser level Bing/Yahoo/..).

I don't see how stating one of the reasons for the sharply declined market share somehow invalidates my earlier statements. I don't think that the underlying causes are relevant to the topic at hand.

> It's not Googles firewall it's government and since no one here seems to like google (for whatever reasons) it doesn't even matter since you simply can use DuckDuck Go (uses Yahoo + some kind of Google Index) or alternatives, they working even in China.

People here are generally against putting too much control into the hands of one single party, hence the name of the add-on. Also, you cannot use DuckDuckGo in mainland China. It's on [the list](https://en.wikipedia.org/wiki/Websites_blocked_in_mainland_China), too.

> The articles says at the very end, it's because the government firewall (great canon).

Nowhere have I claimed that Google is blocking Chinese visitors. People are apparently flocking over to various mirrors, since services hosted under ```*.googleapis.com``` are being blocked. That's all.

**Comment by CHEF-KOCH** [comment]
> I don't see how stating one of the reasons for the sharply declined market share somehow invalidates my earlier statements. I don't think that the underlying causes are relevant to the topic at hand.
> Most chinese cannot visit google, so there are mirrors in china.
> Google's market share in China is small compared to its main competitors (1.7%). 

This is wrong because you can test and see yourself. The main topic is about fonts or .js related stuff wich is already mirrored, I do agree in 2014 and earlier there was problems (non Google related) because China want to pull Google away to force people using Baidu and other things. The Baidu creator is a good friend of a chinese vice-president (Zhang Gaoli, Wang Yang, ..). So no wonder. 


> People here are generally against putting too much control into the hands of one single party, hence the name of the add-on. Also, you cannot use DuckDuckGo in mainland China. It's on the list, too.

To fake something means not you take control over it, I think you not get this, this addon not prevents anything. You can fake whatever you want, the time you put in your data the server gets it, once you have done it it doesn't matter if you never do it again or do it daily since you are in the database. Replacing known _secure_ stuff with stuff _which claims_ to improve something is always critical. See the entre touched encryption history, each time someone touched well _working_ encryption to _improve something_ ended up bad.  

> *.googleapis.com blocked.

It's not blocked it's 'if' only under censorship (because entire CDN was not available), how is this related to your ticket? Google not allows to use other things for several sites, the fallback I mentioned will be used and only this. If you had reads it you would know that this now works. This was only once because my mention things, you can't come up with over 12 year old things and try to say it's blocked or whatever if you never tested it yourself. I daily work with Chinese together and they said it currently works fine, of course I can't speak for everyone. 

If I check the domains like 'sdn.geekzu.org' they not even exist since 2 or more years, they may was created because of the 1+ year old _issue_. You really trust such youngster domain? ...

**Comment by Synzvato** [comment]
> I still see not really any improvement over this, Google needs to handle it and not any unknown mirror page we nothing know about.

The request is not at all about setting up mirror sites for Google Hosted Libraries. It's about preventing requests to large Chinese mirrors, to make people less dependent on them.

> I daily work with Chinese together and they said it currently works fine, of course I can't speak for everyone.

They are quite lucky then. Said bans are still in effect. I have the information first hand (as well) and the reports are less than a week old. In addition to that, please see this third-party analysis:

https://en.greatfire.org/googleapis.com

Google APIs (including ```fonts.googleapis.com```) have been fully blocked for the last 90 days. All of the corresponding domains were last checked under two months ago (on April 28<sup>th</sup> 2016).

> Replacing known secure stuff [...] is always critical. See the entre touched encryption history, each time someone touched well working encryption [...] ended up bad. 

Rolling your own crypto cannot be compared to pointing requests to verified and signed local resources _(using proven encryption algorithms)_. If anything on your machine is malicious and has sufficient rights to modify said files _without_ unsigning the add-on, then we all have much bigger issues to worry about.

**Comment by Synzvato** [comment]
Hi @ivysrono. I have [implemented](https://github.com/Synzvato/decentraleyes/commit/3b13b60bd8a3ce92adc3fa8d266f908669d9e250#diff-c682b42898a8191b14c601ab0b347a97) support for ```sdn.geekzu.org```. However, it seems that its authors are mainly [promoting](https://cdn.geekzu.org/cached.html) other domains such as ```fdn.geekzu.org``` and ```gapis.geekzu.org```. Any thoughts?

**Comment by ivysrono** [comment]
you are right.

thank you very much.

I have updated my URLRedirectorRules with your implement.
